internal/cephfs/controllerserver.go

```diff
@@ -794,7 +794,45 @@ func (cs *ControllerServer) CreateSnapshot(
 	}
 
 	if parentVolOptions.BackingSnapshot {
-		return nil, status.Error(codes.InvalidArgument, "cannot snapshot a snapshot-backed volume")
+		err = store.AddSnapshotBackedVolumeRef(ctx, parentVolOptions, cs.ClusterName, cs.SetMetadata, req.GetSecrets())
```

Comment:
this `if` section get rather large. It is cleaner if you can move it to a helper function.